File: app/src/main/java/io/mgba/Model/Interfaces/IDatabase.java
package io.mgba.Model.Interfaces;


import java.util.List;
import io.mgba.Data.Database.Cheat;
import io.mgba.Data.Database.Game;

public interface IDatabase {
    List<Game> getGamesForPlatform(final int platform);
    List<Game> getFavouritesGames();
    void insert(Game... games);
    void delete();
    void delete(Game game);
    List<Game> queryForGames(String query);
    List<Game> getGames();

    List<Cheat> getGamesCheats(Game key);
    void deleteCheat(Cheat cheat);
    void insertCheat(Cheat cheat);
}

File: app/src/main/java/io/mgba/model/interfaces/IDatabase.kt
package io.mgba.model.interfaces


import io.mgba.data.database.Cheat
import io.mgba.data.database.Game

interface IDatabase {
    val favouritesGames: List<Game>
    val games: List<Game>
    fun getGamesForPlatform(platform: Int): List<Game>
    fun insert(vararg games: Game)
    fun delete()
    fun delete(game: Game)
    fun queryForGames(query: String): List<Game>

    fun getGamesCheats(key: Game): List<Cheat>
    fun deleteCheat(cheat: Cheat)
    fun insertCheat(cheat: Cheat)
}
